pkg/operator/ceph/config/network_test.go

```diff
@@ -31,45 +31,92 @@ import (
-	ctx := &clusterd.Context{
-		Clientset:     clientset,
-		NetworkClient: fakenetclient.NewSimpleClientset().K8sCniCncfIoV1(),
-	}
-
-	//
-	// TEST 1: network definition does not exist
-	//
-	netSelector := map[string]string{
-		"public": "public-network-attach-def",
-	}
-
-	_, err := generateNetworkSettings(ctx, ns, netSelector)
-	assert.Error(t, err)
-
-	//
-	// TEST 2: single dedicated networks
-	//
-	expectedNetworks := []Option{
+	tests := []struct {
+		name        string
+		netSelector map[string]string
+		nw          []networkv1.NetworkAttachmentDefinition
+		expectedNw  []Option
+	}{
+		{
+			name:        "no network definition exists",
+			netSelector: map[string]string{"public": "public-network-attach-def"},
+		},
+		{
+			name:        "only cluster network",
+			netSelector: map[string]string{"cluster": "cluster-network-attach-def"},
+			nw:          []networkv1.NetworkAttachmentDefinition{getClusterNw()},
+			expectedNw: []Option{
+				{
+					Who:    "global",
+					Option: "cluster_network",
+					Value:  "172.18.0.0/16",
+				},
+			},
+		},
 		{
-			Who:    "global",
-			Option: "public_network",
-			Value:  "192.168.0.0/24",
+			name:        "only public network",
+			netSelector: map[string]string{"public": "public-network-attach-def"},
+			nw:          []networkv1.NetworkAttachmentDefinition{getPublicNw()},
+			expectedNw: []Option{
+				{
+					Who:    "global",
+					Option: "public_network",
+					Value:  "192.168.0.0/24",
+				},
+			},
 		},
 		{
-			Who:    "global",
-			Option: "cluster_network",
-			Value:  "192.168.0.0/24",
+			name: "public and cluster network",
+			netSelector: map[string]string{
+				"public":  "public-network-attach-def",
+				"cluster": "cluster-network-attach-def",
+			},
+			nw: []networkv1.NetworkAttachmentDefinition{getPublicNw(), getClusterNw()},
+			expectedNw: []Option{
+				{
+					Who:    "global",
+					Option: "public_network",
+					Value:  "192.168.0.0/24",
+				},
+				{
+					Who:    "global",
+					Option: "cluster_network",
+					Value:  "172.18.0.0/16",
+				},
+			},
 		},
 	}
 
-	// this nad uses whereabouts cni
-	network := &networkv1.NetworkAttachmentDefinition{
+	for _, test := range tests {
+		t.Run(test.name, func(*testing.T) {
+			ctxt := context.TODO()
+			ns := "rook-ceph"
+			clientset := testop.New(t, 1)
+			ctx := &clusterd.Context{
+				Clientset:     clientset,
+				NetworkClient: fakenetclient.NewSimpleClientset().K8sCniCncfIoV1(),
+			}
+
+			if len(test.nw) > 0 {
+				for i := range test.nw {
+					_, err := ctx.NetworkClient.NetworkAttachmentDefinitions(ns).Create(ctxt, &test.nw[i], metav1.CreateOptions{})
+					assert.NoError(t, err)
+				}
+				cephNetwork, err := generateNetworkSettings(ctx, ns, test.netSelector)
+				assert.NoError(t, err)
+				assert.ElementsMatch(t, cephNetwork, test.expectedNw, fmt.Sprintf("networks: %+v", cephNetwork))
+			} else {
+				_, err := generateNetworkSettings(ctx, ns, test.netSelector)
+				assert.Error(t, err)
+			}
```

Comment:
This is far too complex for a unit test. When I suggested to use `t.Run` I did not mean to run templated tests in a loop. Looped tests are fine for simple functions with small inputs and outputs, but not complex cases where we need if-else to check outputs. 

Just run the tests with `t.Run` directly.

```
func TestMyFunc(t *testing.T) {
    t.Run("subtest 1", func(t *testing.T) {
       // subtest stuff
    })
    
    t.Run("subtest 2", func(t *testing.T) {
        // other subtest stuff
    })
}